lxd/storage.go

```diff
@@ -873,3 +882,41 @@ func storagePoolDriversCacheUpdate(cluster *db.Cluster) {
 
 	return
 }
+
+// storageVolumeMount initialises a new storage interface and checks the pool and volume are
+// mounted. If they are not then they are mounted.
+func storageVolumeMount(state *state.State, poolName string, volumeName string, volumeTypeName string, instance device.InstanceIdentifier) error {
+	c, ok := instance.(*containerLXC)
+	if !ok {
+		return fmt.Errorf("Received non-LXC container instance")
+	}
+
+	volumeType, _ := storagePoolVolumeTypeNameToType(volumeTypeName)
+	s, err := storagePoolVolumeAttachInit(state, poolName, volumeName, volumeType, c)
+	if err != nil {
+		return err
+	}
+
+	_, err = s.StoragePoolVolumeMount()
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// storageVolumeUmount unmounts a storage volume on a pool.
+func storageVolumeUmount(state *state.State, poolName string, volumeName string, volumeType int) error {
+	// tomp TODO check about the "default" hardcoded project here.
+	s, err := storagePoolVolumeInit(state, "default", poolName, volumeName, volumeType)
```

Comment:
@stgraber I noticed that the existing code I ported over has the project hardcoded as "default", is this expected here?

Reply:
For custom storage volumes, yes, it is. We don't currently have those namespaced per-project.